src/pages/layout/Layout.tsx

```diff
@@ -17,12 +17,17 @@ import Cookies from "js-cookie";
 
 const Layout = () => {
 
-
-    const setCookie = () => {
+    const setDisclaimerCookie = () => {
         Cookies.set("read_disclaimer", "true", {
           expires: 30,
         });
     };
+
+    const setTranslationCookie = () => {
+        Cookies.set("lang_setting", t("langlink.cookie.current"), {
```

Comment:
here you can use this instead: `i18n.language` (from `const { i18n } = useTranslation();`, see `Chat.tsx` for more example)

Reply:
I do believe i18n.language does refer back to the initialization in the "i18n.ts" where lng is declared.